applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp

```diff
@@ -506,6 +506,61 @@ double ComputePerturbationLocalMachNumber(const Element& rElement, const Process
     return velocity_module / local_speed_of_sound;
 }
 
+template <int Dim, int NumNodes>
+double ComputeUpwindFactor(
+        const double localMachNumberSquared, 
+        const ProcessInfo& rCurrentProcessInfo)
+{
+    // Following Fully Simulataneous Coupling of the Full Potential Equation
+    //           and the Integral Boundary Layer Equations in Three Dimensions
+    //           by Brian Nishida (1996), Equation 2.13
+
+    // read free stream values
+    // default CRITICAL_MACH - 0.99 and default UPWIND_FACTOR_CONSTANT - 1.0
+    const double critical_mach = rCurrentProcessInfo[CRITICAL_MACH];
+    const double upwind_factor_constant = rCurrentProcessInfo[UPWIND_FACTOR_CONSTANT];
+
+    KRATOS_ERROR_IF(localMachNumberSquared < std::numeric_limits<double>::epsilon())
+        << "Error in ComputeUpwindFactor: localMachNumberSquared must be larger than zero." << std::endl;
```

Comment:
After thinking about it, I believe my previous suggestion was not the best solution. In my opinion, it can happen that the local mach number is equal (or almost equal) to 0 in certain elements (close to the stagnation point at the leading edge and close to the trailing edge). For this reason, I think that the appropriate thing to do in this case is to check whether the local mach number is under certain small tolerance and clamp it in order to be able to compute upwind factor. What do you think? Let us discuss later.

```suggestion
    if(localMachNumberSquared < 1e-3){
        localMachNumberSquared = 1e-3
        KRATOS_WARNING("ComputeUpwindFactor") << "localMachNumberSquared is smaller than 1-3 and is being clamped to 1e-3"  <<  std::endl;
}
```